internal/ghworkflow/workflow_auto-render.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// limitations under the License.
+
+package ghworkflow
+
+import "github.com/sapcc/go-makefile-maker/internal/core"
+
+func autoRenderWorkflow(cfg *core.GithubWorkflowConfiguration) error {
+	w := newWorkflow("go-makefile-maker", cfg.Global.DefaultBranch, nil)
+	w.Permissions.Contents = tokenScopeWrite // for peter-evans/create-pull-request to create commit & PR
+
+	w.On.WorkflowDispatch = workflowDispatch{manual: "true"}
+	w.On.PullRequest = pushAndPRTriggerOpts{}
+	w.On.Schedule = []cronExpr{{Cron: "00 07 * * 1"}} // every Monday at 07:00 AM
+
+	j := baseJob("go-makefile-maker")
+	j.addStep(jobStep{
+		Name: "Install go-makefile-maker",
+		Run:  "go install github.com/sapcc/go-makefile-maker@latest",
+	})
+	j.addStep(jobStep{
+		Name: "Run go-makefile-maker",
+		Run:  "go-makefile-maker",
+	})
+
+	j.addStep(jobStep{
+		Name: "Create Pull Request",
+		Uses: "peter-evans/create-pull-request@v4",
+		With: map[string]interface{}{
```

Comment:
Regarding the addition of `assignees` field here. I would prefer if you didn't remove it so that we get a notification for the PR even if we are not watching the repo on GitHub.

With regards to @majewsky comment about not hardcoding names. I would suggest that you include a `githubWorkflow.global.assignees` (or perhaps `maintainers`) field:

```yaml
githubWorkflow:
  global:
    assignees:
      - majewsky
      - SuperSandro2000
```

and then that could be passed to the [`With` field](https://github.com/peter-evans/create-pull-request#action-inputs) of the `peter-evans/create-pull-request@v4` action as a comma-separated string.